network/p2p/dht_test.go

```diff
@@ -0,0 +1,219 @@
+package p2p
```

Comment:
renamed and repurposed the existing `pubsub_test.go` to `dht_test.go`. The old pubsub test is not relevant anymore since we are not using libp2p discovery for the staked network (we user `peermanager` instead).
The old test used a `MockDiscovery` while the new test uses the real DHT discovery.